src/stamps.ts

```diff
@@ -236,45 +256,56 @@ export class StampsManager {
       const stamps = await beeDebug.getAllPostageBatch()
       logger.debug('retrieved stamps', stamps)
 
-      const { amount, ttlMin, depth } = config
+      const { depth, amount, ttlMin, usageThreshold } = config
 
       // Get all usable stamps sorted by usage from most used to least
-      this.usableStamps = filterUsableStampsExtends(stamps)
+      this.usableStampsExtendsTTL = filterUsableStampsExtendsTTL(stamps)
+      this.usableStampsExtendsCapacity = filterUsableStampsExtendsCapacity(stamps, usageThreshold)
 
       if (!this.isBuyingStamp) {
-        if (this.usableStamps.length === 0) {
+        if (this.usableStampsExtendsTTL.length === 0) {
           this.isBuyingStamp = true
           try {
             const { stamp: newStamp } = await buyNewStamp(depth, amount, beeDebug)
 
             // Add the bought postage stamp
-            this.usableStamps.push(newStamp)
+            this.usableStampsExtendsTTL.push(newStamp)
           } finally {
             this.isBuyingStamp = false
           }
         } else {
           await this.verifyUsableStamps(beeDebug, ttlMin, config, amount)
         }
       }
+
+      for (const stamp of this.usableStampsExtendsCapacity) {
+        try {
+          logger.debug(`extending stamp capacity: ${stamp.batchID}`)
+          await topUpStamp(beeDebug, stamp.batchID, (BigInt(stamp.amount) * BigInt(2)).toString())
+          await beeDebug.diluteBatch(stamp.batchID, stamp.depth + 1)
+          logger.info(`capacity extended for stamp ${stamp.batchID}`)
```

Comment:
In the extendsTTL feature, after the topup operation there is a 1 minute wait:

https://github.com/ethersphere/gateway-proxy/blob/master/src/stamps.ts#L280

Topup and dilute are the same in this regard, they both require a short while for the confirmation before Bee accepts and updates to the new values.

So if we have a wait on purpose in the other mode, let's have it here as well, or I am fine not having it in either places if it is not important, but keep it consistent otherwise it is confusing why it only exists for one of the modes.